src/test/java/ch/jalu/collectionbehavior/ListTest.java

```diff
@@ -37,89 +44,182 @@ void assertListSupertypeIsSequencedCollection() {
      * even when null is called on {@link List#contains}.
      */
-    @Test
-    void jdkListOf() {
-        // Is immutable
-        String[] elements = { "a", "b", "c", "d" };
-        List<String> list = List.of(elements);
-        verifyIsImmutable(list, () -> elements[2] = "changed");
-
-        // Implements RandomAccess
-        assertThat(list, instanceOf(RandomAccess.class));
-
-        // May not contain null
-        assertThrows(NullPointerException.class, () -> List.of("a", null, "c"));
-
-        // No null support in methods
-        verifyRejectsNullArgInMethods(list);
+    @TestFactory
+    List<DynamicTest> jdk_List_of(TestInfo testInfo) {
+        return TestsGenerator.forType(testInfo, ListCreator.forArrayBasedType(List::of))
+            .expect(NullSupport.REJECT, true)
+            .mutability(ListModificationBehavior.immutable().alwaysThrows())
+            .createTests();
     }
 
     /**
      * {@link List#copyOf} copies the incoming collection and returns an immutable List;
      * recognizes lists of its own class and returns the same instance instead of unnecessarily copying.
      * Does not support null (not even as argument passed into things like {@link List#contains}).
      */
-    @Test
-    void jdkListCopyOf() {
-        // Is immutable
-        List<String> elements = new ArrayList<>(Arrays.asList("a", "b", "c", "d"));
-        List<String> list = List.copyOf(elements);
-        verifyIsImmutable(list, () -> elements.set(2, "changed"));
-
-        // Implements RandomAccess
-        assertThat(list, instanceOf(RandomAccess.class));
-
-        // May not contain null
-        List<String> listWithNull = Arrays.asList("a", null, "c");
-        assertThrows(NullPointerException.class, () -> List.copyOf(listWithNull));
-
-        // No null support in methods
-        verifyRejectsNullArgInMethods(list);
+    @TestFactory
+    List<DynamicTest> jdk_List_copyOf(TestInfo testInfo) {
+        return TestsGenerator.forType(testInfo, ListCreator.forListBasedType(List::copyOf))
+            .expect(NullSupport.REJECT, true)
+            .mutability(ListModificationBehavior.immutable().alwaysThrows())
+            .skipsWrappingForOwnClass()
+            .createTests();
+    }
 
-        // Does not create new instances if not needed
-        assertThat(List.copyOf(list), sameInstance(list));
+    /**
+     * {@link Arrays#asList} wraps an array into the List interface. Allows to change individual elements
+     * but elements cannot be added or removed. The array passed into it is not copied! Supports null.
+     * Basically wraps an array into a List interface as changes to the array are propagated to the List and
+     * vice versa (changing an entry in the List actually changes the backing array).
+     */
+    @TestFactory
+    List<DynamicTest> jdk_Arrays_asList(TestInfo testInfo) {
+        return TestsGenerator.forType(testInfo, ListCreator.forArrayBasedType(Arrays::asList))
```

Comment:
Check all ListCreators that they still match the type that is intended to be tested